lib/executor/base_executor.go

```diff
@@ -66,6 +83,12 @@ func (bs BaseExecutor) GetConfig() lib.ExecutorConfig {
 	return bs.config
 }
 
+// getNextLocalVUID increments and returns the next VU ID that's specific for
+// this executor (i.e. not global like __VU).
+func (bs BaseExecutor) getNextLocalVUID() uint64 {
+	return atomic.AddUint64(bs.VUIDLocal, 1)
+}
```

Comment:
Is this still needed? Right now we have 3 VU ID values: local to a single scenario (this one), global to a single instance (`__VU`),  and global across instances (added in #2064). I gathered this one from [this comment](https://github.com/k6io/k6/pull/1863#issuecomment-782069940) ("_in the current scenario_, what is the ID number of the VU currently running the iteration"), but I'm wondering how useful that is in practice...

Reply:
I feel like the use case for this VU ID is if you have *multiple* scenarios and need to segment something based on the VUs between them. 

I would argue in most cases the *per scenario* iteration will be sufficient and probably all other cases are some kind of memory optimization that likely is better handled by SharedArray and co.

So I am slightly :+1: for dropping this particular counter, we can always put it back in the future. But I am by no means blocking the PR based on that,

Reply:
Honestly, I would prefer it if we switched to GUIDs instead of incremental IDs, and just have those instead of 3 different types. It would break certain use cases like "run this only on the _first_ VU", but IMO those are workarounds for other features we need like events/hooks. GUIDs would work in both local and distributed contexts, and we would avoid all these counters.

It would be a huge breaking change, but something to think about for v1+?

Reply:
hmm there's nothing stopping you from having GUIDs even now, just have this in the init context:
```js
const vuID = getMyFancyUUID();
```

Integer IDs, especially when we can have them globally unique and mostly sequential so easily, have other benefits that UUIDs don't have, especially when it comes to ad-hoc partitioning of data or work :man_shrugging: so I don't really see a downside in having them here